pkg/ebpftracer/c/headers/common/network.h

```diff
@@ -9,70 +9,79 @@
 } iphdrs;
 
-typedef union  {
-    u32 v4addr;
-    unsigned __int128 v6addr;
-}  __attribute__((packed)) addr_t;
+typedef union {
+    __u8 raw[16];
+    __u32 v4addr;
+    __be32 u6_addr32[4];
+} __attribute__((packed)) addr_t;
 
 typedef struct {
     addr_t saddr;
     addr_t daddr;
-    u16 sport;
-    u16 dport;
-    u16 family;
+    __u16 sport;
+    __u16 dport;
+    __u16 family;
 } __attribute__((packed)) tuple_t;
 
-// network flow events
+union addr {
+    __u8 raw[16];
+    __be32 ipv6[4];
+    __be32 ipv4;
+} __attribute__((__packed__));
 
-typedef struct netflow {
-    u32 host_pid;
-    u8 proto;
-    tuple_t tuple;
-} __attribute__((__packed__)) netflow_t;
+typedef struct process_identity {
+    __u32 pid;
+    __u64 pid_start_time;
+    __u64 cgroup_id;
+    __u8 comm[TASK_COMM_LEN];
+} __attribute__((__packed__)) process_identity_t;
 
-statfunc netflow_t invert_netflow(netflow_t flow)
-{
-    tuple_t inverted_tuple = {
-        .saddr = flow.tuple.daddr,
-        .daddr = flow.tuple.saddr,
-        .sport = flow.tuple.dport,
-        .dport = flow.tuple.sport,
-        .family = flow.tuple.family,
-    };
-    netflow_t res = {
-       .host_pid = flow.host_pid,
-       .proto = flow.proto,
-       .tuple = inverted_tuple,
-    };
-    return res;
-}
+struct traffic_summary {
+    size_t rx_packets;
+    size_t rx_bytes;
+
+    size_t tx_packets;
+    size_t tx_bytes;
+
+    __u64 last_packet_ts;
+    // In order for BTF to be generated for this struct, a dummy variable needs to
+    // be created.
+} __attribute__((__packed__)) traffic_summary_dummy;
 
-typedef struct netflowvalue {
-    s64 active;                             // count active connections
-    u64 last_update;                        // last time this flow was updated
-    u64 tx_bytes;                           // total bytes sent
-    u64 rx_bytes;                           // total bytes received
-    u64 tx_packets;                         // total packets sent
-    u64 rx_packets;                         // total packets received
-} __attribute__((__packed__)) netflowvalue_t;
+struct ip_key {
+    struct process_identity process_identity;
 
-// netflowmap (keep track of network flows)
+    tuple_t tuple;
+    __u8 proto;
+
+    // In order for BTF to be generated for this struct, a dummy variable needs to
+    // be created.
+} __attribute__((__packed__)) ip_key_dummy;
 
 struct {
-    __uint(type, BPF_MAP_TYPE_LRU_HASH);
-    __uint(max_entries, 65535);             // simultaneous network flows being tracked
-    __type(key, netflow_t);                 // the network flow ...
-    __type(value, netflowvalue_t);          // ... linked to flow stats
-} netflowmap SEC(".maps");                  // relate sockets and tasks
+    __uint(type, BPF_MAP_TYPE_ARRAY_OF_MAPS);
+    __uint(max_entries, 2);
+    __type(key, int);
+    __array(
+        values, struct {
+            __uint(type, BPF_MAP_TYPE_LRU_HASH);
+            __uint(max_entries, 10240);
```

Comment:
Before it was `65535` possible netflows.  I suspect that large nodes could have more than `10240` active flows.
But in anyway it could be really useful to later export such maps current entries counts as a metric so we can track these sizes.